src/librustdoc/json/mod.rs

```diff
@@ -172,7 +172,11 @@ impl<'tcx> FormatRenderer<'tcx> for JsonRenderer<'tcx> {
     /// implementations filled out before they're inserted.
     fn item(&mut self, item: clean::Item) -> Result<(), Error> {
         // Flatten items that recursively store other items
-        item.kind.inner_items().for_each(|i| self.item(i.clone()).unwrap());
+        // We skip local blanket implementations, as we'll have already seen the actual generic
+        // impl, and the generated ones don't need documenting.
```

Comment:
This doesn't seem like necessarily a good way to solve the problem. I assume users will want to see all the impls for a type, not just the non-local-blanket impls?

Plus, it just seems like covering up the underlying problem, which is a mismatch between the visibility etc. that rustdoc generates for blanket impls and what it generates for regular impls.

Reply:
This still generates the impl itself below this check, this just skips its children, which we know will have already been documented in this case. I think fixing the visibility / self ty / return type would also be nice, but I think 'don't document the same thing twice' is a fine solution in its own right. If you want, I can mark this a HACK/FIXME as a temporary solution till the inconsistency is otherwise fixed.

Reply:
Hmm... could you add a HACK/FIXME for now?

Reply:
Made the comment a FIXME explaining why it's necessary and indicating it can be removed if the mismatch is fixed